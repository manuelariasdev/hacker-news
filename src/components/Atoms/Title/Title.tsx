import React from 'react'
import { StyledTitle } from './titleStyles'
import type { Font, Size } from '../../../types'

export interface TitleProps {
    size?: Size;
    font?: Font;
    children?: React.ReactNode;
  }

export const Title = ({ children, font = 'primary', size = 'md' }: TitleProps) => {
  return (
    <StyledTitle font={font} size={size}>
      {children}
    </StyledTitle>
  )
}
