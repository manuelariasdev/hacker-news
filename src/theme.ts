
import { DefaultTheme } from 'styled-components'

const theme: DefaultTheme = {
  borderRadius: {
    sm: '3px',
    md: '6px',
    lg: '15px',
    xl: '50px'
  },
  boxShadow: {
    sm: ' 0 1px 4px 0 rgba(0, 21, 41, 0.12)'
  },
  fontSize: {
    xs: '0.68rem', 
    sm: '0.875rem', 
    md: '16px',
    lg: '1.25rem', 
    xl: '1.75rem'
  },
  letterSpacing: {
    md: 0.25
  },
  lineHeight: {
    xs: '1rem',
    sm: '1.25rem',
    md: '16px',
    lg: '1.25rem',
    xl: '1.75rem'
  },
  transition: 'all 700ms cubic-bezier(0.15, 1, 0.20, 1)'
}

export default theme
