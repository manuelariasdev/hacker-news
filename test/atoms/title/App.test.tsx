/**
 * @jest-environment jsdom
 */
import * as React from "react";
import { ThemeProvider } from 'styled-components'
import theme from '../../../theme';
import { render } from '@testing-library/react'
import { Title } from '../../../src/components/Atoms/Title/Title';

describe('Title <Title/>', function () {


        test('debe de hacer match con el snapshot Title', () => {
          
            const {container} = render(<ThemeProvider theme={theme}><Title font="secondary" size = 'md' /> </ThemeProvider>)
            expect( container ).toMatchSnapshot();
         })
      
   
});