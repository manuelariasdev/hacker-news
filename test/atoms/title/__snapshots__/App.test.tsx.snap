// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title <Title/> debe de hacer match con el snapshot Title 1`] = `
<div>
  <p
    class="sc-bczRLJ kMVQkp"
  />
   
</div>
`;
